doc/cephfs/nfs.rst

```diff
@@ -79,3 +79,219 @@ Current limitations
+
+
+Deploy Minikube Cluster::
+
+        minikube start --vm-driver=kvm2
+
+.. note:: Add --memory and --cpus options to provide custom RAM and CPU values for the VM.
+
+List the running pods::
+
+        kubectl get pods --all-namespaces
+
+Rook Setup And Cluster Deployment
+---------------------------------
+
+Clone the rook repository::
+
+        git clone https://github.com/rook/rook.git
+
+Deploy the Rook Operator::
+
+        cd cluster/examples/kubernetes/ceph
+        kubectl create -f common.yaml
+        kubectl create -f operator.yaml
+
+.. note:: Nautilus release or latest Ceph image should be used.
+
+Before proceding check if the pods are running::
+
+        kubectl -n rook-ceph get pod
+
+.. note:: For any troubleshooting on any pod use::
+          kubectl describe -n rook-ceph pod <pod-name>
+
+In cluster-test.yaml file set **allowMultiplePerNode** to **true** under mon. As we
+will be using 3 monitors. If using minkube cluster change the **dataDirHostPath**
+to **/data/rook**. This is to make sure data persists across reboots.
+
+Deploy the ceph cluster::
+
+        kubectl create -f cluster-test.yaml
+
+Now check if you have one Ceph monitor, manager and OSD running.
+
+To interact with Ceph Daemons, let's deploy toolbox::
+
+        kubectl create -f ./toolbox.yaml
+
+Exec into the rook-ceph-tools pod::
+
+        kubectl -n rook-ceph exec -it $(kubectl -n rook-ceph get pod -l "app=rook-ceph-tools" -o jsonpath='{.items[0].metadata.name}') bash
+
+Check Ceph cluster status::
+
+        [root@minikube /]# ceph -s
+           cluster:
+                id:     3a30f44c-a9ce-4c26-9f25-cc6fd23128d0
+                health: HEALTH_OK
+
+           services:
+                mon: 1 daemons, quorum a (age 14m)
+                mgr: a(active, since 13m)
+                osd: 1 osds: 1 up (since 13m), 1 in (since 13m)
+
+           data:
+                pools:   0 pools, 0 pgs
+                objects: 0 objects, 0 B
+                usage:   5.0 GiB used, 11 GiB / 16 GiB avail
+                pgs:
+
+Scale out monitors::
+
+        [root@minikube /]# ceph orchestrator mon update 3
+        Updated mon count to 3
+
+It takes few minutes for the monitor daemon to come up. Check the status with::
+
+        [root@minikube /]# ceph orchestrator service ls
+        mgr a minikube rook-ceph-mgr-a-55964c5888-w9sbt None None
+        mon a minikube rook-ceph-mon-a-579c967b98-dv7tw None None
+        mon b minikube rook-ceph-mon-b-784b5b5db5-dpnpf None None
+        mon c minikube rook-ceph-mon-c-585d8d5f6b-25v2l None None
+        osd 0 minikube rook-ceph-osd-0-b9666466-r7678 None None
+
+Create a Ceph File System
+-------------------------
+Using ceph-mgr volumes module, we will create a ceph file system::
+
+        [root@minikube /]# ceph fs volume create myfs
+
+By default replicated size for OSD is 3. Since we are using only one OSD. It can cause error. Let's fix this up by setting replicated size to 1.::
```

Comment:
Maybe add a note here that you should ONLY try this at home (i.e. not in production)